tartufo/scanner.py

```diff
@@ -38,26 +43,23 @@
 
 BASE64_REGEX = re.compile(r"[A-Z0-9=+/_-]+", re.IGNORECASE)
 HEX_REGEX = re.compile(r"[0-9A-F]+", re.IGNORECASE)
+OUTPUT_SEPARATOR = "~~~~~~~~~~~~~~~~~~~~~"
```

Comment:
This seems slightly flaky -- I guess the same complaint could be made about the two REGEX expressions too -- in the sense that these definitions are integral to the behavior of the class but defined outside it and silently consumed.

I presume you moved OUTPUT_SEPARATOR out to reduce the size of each object, but is it feasible to make these class variables? This would be better encapsulation, but if the actual python implementation still ends up duplicating the boilerplate in every instance, that's obviously not a win.

Reply:
I tried a very simple test, and it appears that class variables do not appear in the output of pickled instances. FWIW.

Reply:
> This seems slightly flaky -- I guess the same complaint could be made about the two REGEX expressions too -- in the sense that these definitions are integral to the behavior of the class but defined outside it and silently consumed.
> 
> 
> 
> I presume you moved OUTPUT_SEPARATOR out to reduce the size of each object, but is it feasible to make these class variables? This would be better encapsulation, but if the actual python implementation still ends up duplicating the boilerplate in every instance, that's obviously not a win.

This is absolutely moved out to reduce the size of the Issue object (both serialized and on the heap), but I think it's totally valid feedback that it should be static class variables rather than global.  I need to learn the Python syntax for that and make the change.  As long as it doesn't end up increasing the size of the serialized object, I'll make this change.  Thanks!